lib/Interpreter/DeclUnloader.cpp

```diff
@@ -711,10 +736,49 @@ bool DeclUnloader::VisitRedeclarable(clang::Redeclarable<T>* R, DeclContext* DC)
 
   bool DeclUnloader::VisitNamespaceDecl(NamespaceDecl* NSD) {
     // NamespaceDecl: NamedDecl, DeclContext, Redeclarable
-    bool Successful = VisitDeclContext(NSD);
-    Successful &= VisitRedeclarable(NSD, NSD->getDeclContext());
+
+    // When unloading a NamedDecl within an inline namespace within std the
+    // NamedDecl needs to be removed from the parent std as well.
+    // Not sure why, but test/CodeUnloading/NameSpaces.C demonstrates
```

Comment:
I suspect Vassil is pointing the behavior is not changed with the PR (as expected since the change is specific to std).

> And doesn't the .undo 2 mean there is no longer an f, so the error is correct?

Wouldn't the .undo remove the decl for A::[__BBB]::f and thus, shouldn't it be claiming that A::f does not exist rather than failing to find the symbol?

Cheers,
Philippe.